cpp/include/cudf/io/orc.hpp

```diff
@@ -0,0 +1,832 @@
+   * @return Builder to build reader options.
+   */
+  static orc_reader_options_builder builder(source_info const& src);
+
+  /**
+   * @brief Returns source info.
+   */
+  source_info const& get_source() const { return _source; }
+
+  /**
+   * @brief Returns names of the columns to read.
+   */
+  std::vector<std::string> const& get_columns() const { return _columns; }
+
+  /**
+   * @brief Returns list of individual stripes to read.
+   */
+  std::vector<size_type> const& get_stripes() const { return _stripes; }
+
+  /**
+   * @brief Returns number of rows to skip from the start.
+   */
+  size_type get_skip_rows() const { return _skip_rows; }
+
+  /**
+   * @brief Returns number of row to read.
+   */
+  size_type get_num_rows() const { return _num_rows; }
+
+  /**
+   * @brief Whether to use row index to speed-up reading.
+   */
+  bool is_enabled_use_index() const { return _use_index; }
+
+  /**
+   * @brief Whether to use numpy-compatible dtypes.
+   */
+  bool is_enabled_use_np_dtypes() const { return _use_np_dtypes; }
+
+  /**
+   * @brief Returns timestamp type to which timestamp column will be cast.
+   */
+  data_type get_timestamp_type() const { return _timestamp_type; }
+
+  /**
+   * @brief Whether to convert decimals to float64.
+   */
+  bool is_enabled_decimals_as_float64() const { return _decimals_as_float64; }
+
+  /**
+   * @brief Returns whether decimal scale is inferred or forced to have limited fractional digits.
+   */
+  size_type get_forced_decimals_scale() const { return _forced_decimals_scale; }
+
+  // Setters
+
+  /**
+   * @brief Sets names of the column to read.
+   *
+   * @param col_names Vector of column names.
+   */
+  void set_columns(std::vector<std::string> col_names) { _columns = std::move(col_names); }
+
+  /**
+   * @brief Sets list of individual stripes to read.
+   *
+   * @param stripes Vector of individual stripes.
+   */
+  void set_stripes(std::vector<size_type> stripes)
+  {
+    CUDF_EXPECTS(stripes.empty() or (_skip_rows == 0), "Can't set stripes along with skip_rows");
+    CUDF_EXPECTS(stripes.empty() or (_num_rows == -1), "Can't set stripes along with num_rows");
+    _stripes = std::move(stripes);
+  }
+
+  /**
+   * @brief Sets number of rows to skip from the start.
+   *
+   * @param rows Number of rows.
+   */
+  void set_skip_rows(size_type rows)
+  {
+    CUDF_EXPECTS(rows == 0 or _stripes.empty(), "Can't set both skip_rows along with stripes");
+    _skip_rows = rows;
+  }
+
+  /**
+   * @brief Sets number of row to read.
+   *
+   * @param nrows Number of rows.
+   */
+  void set_num_rows(size_type nrows)
+  {
+    CUDF_EXPECTS(nrows == -1 or _stripes.empty(), "Can't set both num_rows along with stripes");
+    _num_rows = nrows;
+  }
+
+  /**
+   * @brief Enable/Disable use of row index to speed-up reading.
+   *
+   * @param use Boolean value to enable/disable row index use.
+   */
+  void enable_use_index(bool use) { _use_index = use; }
```

Comment:
If we want consistency with the `set` convention, we could prepend `should` to the underlying
```suggestion
  void set_should_use_index(bool use) { _should_use_index = use; }
```

Reply:
There was a discussion where we decided to go with enable and is_eanbled for boolean options. https://github.com/rapidsai/cudf/pull/6051#discussion_r481807727